rts/lra/jaxrs/src/main/java/io/narayana/lra/filter/ClientLRARequestFilter.java

```diff
@@ -22,20 +22,40 @@
 package io.narayana.lra.filter;
 
 import io.narayana.lra.Current;
-
 import jakarta.ws.rs.client.ClientRequestContext;
 import jakarta.ws.rs.client.ClientRequestFilter;
+import jakarta.ws.rs.core.MultivaluedMap;
 import jakarta.ws.rs.ext.Provider;
 
+import java.net.URI;
+
 import static org.eclipse.microprofile.lra.annotation.ws.rs.LRA.LRA_HTTP_CONTEXT_HEADER;
 
 @Provider
 public class ClientLRARequestFilter implements ClientRequestFilter {
+
     @Override
     public void filter(ClientRequestContext context) {
-        if (Current.peek() != null) {
-            context.setProperty(LRA_HTTP_CONTEXT_HEADER, Current.peek());
+        MultivaluedMap<String, Object> headers = context.getHeaders();
+
+        if (headers.containsKey(LRA_HTTP_CONTEXT_HEADER)) {
+            // LRA context is explicitly set
+            return;
+        }
+
+        URI lraId = Current.peek();
```

Comment:
Issue JBTM-3552 (Do not rely on thread locals for propagating LRA context) is because you cannot know which thread is used to execute outgoing requests: it might be the same thread used to run the incoming request but it could be on a thread that was used for a completely different incoming request. In the latter case this peek will return the wrong LRA (using the `activeLRACache` only catches requests that have finished).

Reply:
@marcosgopen i.e. I think this change suffers from the same issue, although it should reduce the possibility of getting the wrong context (but transactions must be reliable).

Reply:
Thanks @mmusgrov for your review. IIUC we still need to solve [JBTM-3552](https://issues.redhat.com/browse/JBTM-3552). @xstefank Do you still wish to integrate this PR or should we wait until 3552 gets fixed?

Reply:
The referenced issue for this PR is JBTM-3749 so I am not sure it is necessary for this PR to solve JBTM-3552 too.